src/App.tsx

```diff
@@ -1,25 +1,109 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { TodoList } from './components/TodoList';
+
+export const App: React.FC = () => {
+  const getUser = (param: number | string): User | null => {
+    const foundUser
+    = typeof param === 'number'
+      ? usersFromServer.find(user => user.id === param)
+      : usersFromServer.find(user => user.name === param);
+
+    return foundUser || null;
+  };
+
+  const todos: Todo[] = todosFromServer.map(todo => ({
+    ...todo,
+    user: getUser(todo.userId),
+  }));
+
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [newTodoUserName, setNewTodoUserName] = useState('');
+  const [renderedTodos, setRenderedTodos] = useState(todos);
+  const [needCheck, setNeedCheck] = useState(false);
+
+  const handleChangeInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+    const rightValue = value.trim().length > 0
```

Comment:
You don't have to check for this condition actually, if the length is 0, replace will leave it. So this should work also:

```js
const rightValue = value.replace(/[^a-zA-Z0-9 ]/g, '');